lib/soapy_cake/response.rb

```diff
@@ -29,6 +29,15 @@ def to_xml
       (empty? ? [] : [body.to_s]).to_enum
     end
 
+    def total_rows
+      body.to_s.scan(%r{<row_count>(\d+)<\/row_count>}).flatten.first.to_i
+    end
+
+    def number_of_elements
+      depth = short_response ? SHORT_ELEMENT_DEPTH : ELEMENTS_DEPTH
+      sax.at_depth(depth).count
```

Comment:
This is really slow because it parses the whole XML again. Do we even need it (we already count the elements in AdminBatched)?